go-controller/pkg/libovsdb/libovsdb.go

```diff
@@ -203,6 +203,7 @@ func NewNBClientWithConfig(cfg config.OvnAuthConfig, promRegistry prometheus.Reg
 		nbdb.LoadBalancerTable:  {{Columns: []model.ColumnKey{{Column: "name"}}}},
 		nbdb.LogicalSwitchTable: {{Columns: []model.ColumnKey{{Column: "name"}}}},
 		nbdb.LogicalRouterTable: {{Columns: []model.ColumnKey{{Column: "name"}}}},
+		nbdb.QoSTable:           {{Columns: []model.ColumnKey{{Column: "external_ids", Key: types.PrimaryIDKey}}}},
```

Comment:
looks like address set type is also missing...why?

Reply:
so I cannot do LogicalRouterPolicy yet because the rest of the LRPs do not yet have externalIDs so that will need to be a refactor in a separate PR change. However when we do add new policies we should start using externalIDs so that the amount of conversion we have to do is less

re: address-set... hmm no idea I'm pretty sure Nadia converted all of them to use externalIDs.. @npinaeva do you know why AddressSets are not in this list for client side indexing using primaryIDs?

Reply:
ok so the rule of thumb here is we can only add client indexes once all objects have been converted to db ids...makes sense. Let's wait for @npinaeva to give us the status on address sets.

Reply:
yeah my guess is going to be that so far all address-sets have unique names and name is the indexing we use today which probably suffices, but let's wait to see what Nadia says because the whole point of changing to use indexes was for this..

Reply:
that is because address set name is a server-side index and it is based on primary id https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/ovn/address_set/address_set.go#L263-L265
all address set operations are done via address set factory that makes sure the name is always set.
For port groups it is the same https://github.com/ovn-org/ovn-kubernetes/blob/master/go-controller/pkg/libovsdb/util/port_group.go#L10
anything that doesn't have a primaryID-based server-side index should leverage client side indexing like you did here.

Reply:
oh! that's good to know; I think we should have docs for indexing OR even a broader set of docs around libovsdb in our contributor docs; I'll open an issue to track it

1. client side indexing
2. server side indexing
3. how ovnkube uses libovsdb
4. moving forward all new OVN Constructs must leverage dbindexes
5. details of indexing nuances of each ovn object in ovnkube etc


We'll see if we can find cycles during our documentation-workshop-friday if we can plan such a thing :D

opened https://github.com/ovn-org/ovn-kubernetes/issues/4362 and resolving this comment.